Using the aspire 9.0.0 SDK with a prerelease nuget version expression fails

```xml
<Project Sdk="Microsoft.NET.Sdk;Microsoft.NET.Sdk.Publish">
	
	<Sdk Name="Aspire.AppHost.Sdk" Version="9.0.0" />

	<PropertyGroup>
		<OutputType>Exe</OutputType>
		<TargetFramework>net8.0</TargetFramework>
		<ImplicitUsings>enable</ImplicitUsings>
		<Nullable>enable</Nullable>
		<UserSecretsId>ff7ad17b-2d94-4ac9-a472-b1b236c733c7</UserSecretsId>
		<IsPublishable>true</IsPublishable>
		<NoWarn>NETSDK1023;NETSDK1057</NoWarn>
		<UseAppHost>false</UseAppHost>
	</PropertyGroup>

	<ItemGroup>
		<PackageReference Include="Aspire.Hosting.AppHost" Version="9.*-*" />
	</ItemGroup>

</Project>
```

```
dotnet build C:\dev\git\AspireApps10\AspireApps10.sln /property:GenerateFullPaths=true /consoleloggerparameters:NoSummary /p:Configuration=Debug /p:Platform="Any CPU" 
  Determining projects to restore...
C:\Users\davifowl\.nuget\packages\aspire.apphost.sdk\9.0.0\Sdk\Sdk.targets(80,20): error MSB4184: The expression "[MSBuild]::VersionLessThan(9.*-*, 8.2.0)" cannot be evaluated. Version string was not in a correct format. [C:\dev\git\AspireApps10\samples\AspireApp1.DevHost\AspireApp1.DevHost.csproj]
```

--- a/src/Aspire.AppHost.Sdk/SDK/Sdk.in.targets
+++ b/src/Aspire.AppHost.Sdk/SDK/Sdk.in.targets
@@ -77,7 +77,12 @@
       <_AppHostVersion>%(_AppHostPackageReference.Version)</_AppHostVersion>
     </PropertyGroup>
 
-    <PropertyGroup Condition="'$(_AppHostVersion)' != '' and $([MSBuild]::VersionLessThan('$(_AppHostVersion)', '8.2.0'))">
+    <PropertyGroup Condition="'$(_AppHostVersion)' != ''">
+      <!-- Check if the version is in the format major.minor.patch -->
+      <_IsValidVersion>$([System.Text.RegularExpressions.Regex]::IsMatch('$(_AppHostVersion)', '^\d+\.\d+\.\d+$'))</_IsValidVersion>
+    </PropertyGroup>
+
+    <PropertyGroup Condition="'$(_AppHostVersion)' != '' and '$(_IsValidVersion)' == 'true' and $([MSBuild]::VersionLessThan('$(_AppHostVersion)', '8.2.0'))">
       <!-- If we find the version to Aspire.Hosting.AppHost package but it is lower than 8.2.0, then we fall back
       to use the Dashboard and DCP packages that match the version of the installed workload for backwards compatibility.
       This results in the same behavior that we had before moving Dashboard and DCP out of the workload, since the version

--- a/tests/Aspire.Workload.Tests/BuildAndRunTemplateTests.cs
+++ b/tests/Aspire.Workload.Tests/BuildAndRunTemplateTests.cs
@@ -158,6 +158,32 @@ public partial class BuildAndRunTemplateTests : WorkloadTestsBase
         }
     }
 
+    [Theory]
+    [InlineData("9.*-*")]
+    [InlineData("[9.0.0]")]
+    public async Task CreateAndModifyAspireAppHostTemplate(string version)
+    {
+        string id = GetNewProjectId(prefix: $"aspire_apphost_{version.Replace("*", "wildcard").Replace("[", "").Replace("]", "")}");
+        await using var project = await AspireProject.CreateNewTemplateProjectAsync(id, "aspire-apphost", _testOutput, buildEnvironment: BuildEnvironment.ForDefaultFramework);
+
+        ModifyProjectFile(project, version);
+
+        await project.BuildAsync();
+        await project.StartAppHostAsync();
+
+        static void ModifyProjectFile(AspireProject project, string version)
+        {
+            var projectName = Directory.GetFiles(project.AppHostProjectDirectory, "*.csproj").FirstOrDefault();
+            Assert.False(string.IsNullOrEmpty(projectName));
+
+            var projectContents = File.ReadAllText(projectName);
+
+            var modifiedContents = AppHostVersionRegex().Replace(projectContents, $@"<PackageReference Include=""Aspire.Hosting.AppHost"" Version=""{version}"" />");
+
+            File.WriteAllText(projectName, modifiedContents);
+        }
+    }
+
     [GeneratedRegex(@"<PackageReference\s+Include=""Aspire\.Hosting\.AppHost""\s+Version=""([^""]+)""\s+/>")]
     private static partial Regex AppHostVersionRegex();
 }

--- a/tests/Aspire.Workload.Tests/BuildAndRunTemplateTests.cs
+++ b/tests/Aspire.Workload.Tests/BuildAndRunTemplateTests.cs
@@ -164,16 +164,15 @@ public partial class BuildAndRunTemplateTests : WorkloadTestsBase
     public async Task CreateAndModifyAspireAppHostTemplate(string version)
     {
         string id = GetNewProjectId(prefix: $"aspire_apphost_{version.Replace("*", "wildcard").Replace("[", "").Replace("]", "")}");
-        await using var project = await AspireProject.CreateNewTemplateProjectAsync(id, "aspire-apphost", _testOutput, buildEnvironment: BuildEnvironment.ForDefaultFramework);
+        await using var project = await AspireProject.CreateNewTemplateProjectAsync(id, "aspire-apphost", _testOutput, buildEnvironment: BuildEnvironment.ForDefaultFramework, addEndpointsHook: false);
 
         ModifyProjectFile(project, version);
 
-        await project.BuildAsync();
-        await project.StartAppHostAsync();
+        await project.BuildAsync(workingDirectory: project.RootDir);
 
         static void ModifyProjectFile(AspireProject project, string version)
         {
-            var projectName = Directory.GetFiles(project.AppHostProjectDirectory, "*.csproj").FirstOrDefault();
+            var projectName = Directory.GetFiles(project.RootDir, "*.csproj").FirstOrDefault();
             Assert.False(string.IsNullOrEmpty(projectName));
 
             var projectContents = File.ReadAllText(projectName);

